resources/views/components/backend/marks_allocation.blade.php

```diff
@@ -1,55 +1,99 @@
+                    const value = Number(this.marks_allocation[key]);
+                    if (value) {
+                        hasValue = true; // At least one field has a value
+                    }
+                    if (this.isValidInput(key).valid) {
+                        total += (value || 0);
+                    }
+                });
+
+                this.isEmpty = !hasValue; // Update isEmpty based on presence of values
+                this.totalIsValid = total == 100;
+                return total;
+            }
+          }">
+  <div class="py-2 px-3 my-2" style="border: 1px solid rgb(207, 207, 207); border-radius:5px">
+      <div class="div pt-3">
+          <label for="drop1">Marks Allocation</label>
+      </div>
+      <hr>
+      <div class="row pb-2">
+          <div class="row d-flex align-items-center">
+              <div class="col-md-3 col-3">
+                  <label>Practicals</label>
+              </div>
+              <div class="input-group mb-3 col ms-3">
+                  <input type="number" class="form-control" aria-describedby="practicals" :class="!isValidInput('practicals').valid ? 'is-invalid' : ''" placeholder="20" x-model.number="marks_allocation.practicals" wire:model.defer="marks_allocation.practicals" min="0" max="100">
+                  <span class="input-group-text" id="practicles">%</span>
+                  <div class="invalid-feedback" x-text="isValidInput('practicals').message"></div>
+              </div>
+          </div>
+          <div class="row d-flex align-items-center">
+              <div class="col-md-3 col-3">
+                  <label>Project</label>
+              </div>
+              <div class="input-group mb-3 col ms-3">
+                  <input type="number" class="form-control" aria-describedby="project" :class="!isValidInput('project').valid ? 'is-invalid' : ''" placeholder="20" x-model.number="marks_allocation.project" wire:model.defer="marks_allocation.project" min="0" max="100">
+                  <span class="input-group-text" id="project">%</span>
+                  <div class="invalid-feedback" x-text="isValidInput('project').message"></div>
+              </div>
+          </div>
+          <div class="row d-flex align-items-center">
+              <div class="col-md-3 col-3">
+                  <label style="white-space: nowrap;">Mid-Exam</label>
+              </div>
+              <div class="input-group mb-3 col ms-3">
+                  <input type="number" class="form-control" aria-describedby="mid_exam" :class="!isValidInput('mid_exam').valid ? 'is-invalid' : ''" placeholder="20" x-model.number="marks_allocation.mid_exam" wire:model.defer="marks_allocation.mid_exam" min="0" max="100">
+                  <span class="input-group-text" id="mid_exam">%</span>
+                  <div class="invalid-feedback" x-text="isValidInput('mid_exam').message"></div>
+              </div>
+          </div>
```

Comment:
Will you be able to use a blade for a loop with this? Most of the codes are repetitive content

Reply:
Of course it is possible. I just kept hasantha's design as it is and binded data